### PR TITLE
fixed an issue where the context was not updated when logout was called

### PR DESCRIPTION
**Pull request by mplayer78**

### CODE REVIEW

File: chowpronto_frontend/src/hooks/useAuth.js [hunk 1]
```diff
@@ -3,9 +3,9 @@ import UserContext from "../state/UserContext";
 import useError from "./useError";
 
 function useAuth() {
-  const context = useContext(UserContext);
-  const user = context?.user;
-  const setUser = context?.setUser;
+  const { user, setUser } = useContext(UserContext);
+  // const user = context?.user;
```
Comment (kosenchiha) [inline chowpronto_frontend/src/hooks/useAuth.js:7]: Commented out code should be removed

File: chowpronto_frontend/src/hooks/useAuth.js [hunk 2]
```diff
@@ -60,8 +60,9 @@ function useAuth() {
       );
   }
   async function logout() {
+    setUserDetailsToContext({});
+    console.log("user", user);
```
Comment (kosenchiha) [inline chowpronto_frontend/src/hooks/useAuth.js:64]: console.log  should be removed

File: chowpronto_frontend/src/hooks/useAuth.js [hunk 3]
```diff
@@ -101,6 +102,7 @@ function useAuth() {
     }
   }
   function setUserDetailsToContext(userDetails) {
+    console.log("setUserDetailsToContext called");
```
Comment (kosenchiha) [inline chowpronto_frontend/src/hooks/useAuth.js:105]: console.log :)

File: chowpronto_frontend/src/patron/pages/menu/components/MenuSidebar.js [hunk 4]
```diff
@@ -14,6 +14,7 @@ import { CheckoutButton } from "../../../../shared_components/CheckoutButton";
 const MenuSidebar = () => {
   const { getUser, logout } = useAuth();
   const user = getUser();
+  console.log("user", user);
```
Comment (kosenchiha) [inline chowpronto_frontend/src/patron/pages/menu/components/MenuSidebar.js:17]: again :)

File: chowpronto_frontend/src/state/reducer.js [hunk 5]
```diff
@@ -46,11 +46,11 @@ export function reducer(state, action) {
...
-    case "set_user":
-      return {
-        ...state,
-        userDetails: { ...action.userDetails },
-      };
+    // case "set_user":
+    //   return {
+    //     ...state,
+    //     userDetails: { ...action.userDetails },
+    //   };
```
Comment (kosenchiha) [inline chowpronto_frontend/src/state/reducer.js:53]: Commented out code should be removed

File: chowpronto_frontend/src/hooks/useAuth.js (hunk 2 above)
Comment (mplayer78) [reply chowpronto_frontend/src/hooks/useAuth.js]: Done

File: chowpronto_frontend/src/patron/pages/menu/components/MenuSidebar.js (hunk 4 above)
Comment (mplayer78) [reply chowpronto_frontend/src/patron/pages/menu/components/MenuSidebar.js:17]: Done

File: chowpronto_frontend/src/state/reducer.js (hunk 5 above)
Comment (mplayer78) [reply chowpronto_frontend/src/state/reducer.js]: Done

File: chowpronto_frontend/src/hooks/useAuth.js (hunk 3 above)
Comment (mplayer78) [reply chowpronto_frontend/src/hooks/useAuth.js]: Done

File: chowpronto_frontend/src/hooks/useAuth.js (hunk 1 above)
Comment (mplayer78) [reply chowpronto_frontend/src/hooks/useAuth.js]: Done

**Pull request merged**